app/javascript/controllers/image_upload_controller.js

```diff
@@ -1,56 +1,80 @@
 import { Controller } from "@hotwired/stimulus"
```

Comment:
I did a bunch of rework of this whole stimulus controller. It's probably easier to view it as the finished file rather than the diff